pkg/forwarder/blocked_endpoints.go

```diff
@@ -29,32 +36,44 @@ func newBlockedEndpoints() *blockedEndpoints {
 	return &blockedEndpoints{errorPerEndpoint: make(map[string]*block)}
 }
 
-func (e *blockedEndpoints) block(endpointName string) {
+func (e *blockedEndpoints) close(endpoint string) {
 	e.m.Lock()
 	defer e.m.Unlock()
 
 	var b *block
-	if knownBlock, ok := e.errorPerEndpoint[endpointName]; ok {
+	if knownBlock, ok := e.errorPerEndpoint[endpoint]; ok {
 		b = knownBlock
 	} else {
 		b = &block{}
 	}
-	b.nbError++
 
-	newInterval := time.Duration(b.nbError) * blockInterval
-	if newInterval > maxRetryInterval {
-		newInterval = maxBlockInterval
+	b.nbError++
+	if b.nbError > maxErrors {
+		b.nbError = maxErrors
 	}
-	b.until = time.Now().Add(newInterval)
+	b.until = time.Now().Add(GetBackoffDuration(b.nbError))
 
-	e.errorPerEndpoint[endpointName] = b
+	e.errorPerEndpoint[endpoint] = b
 }
 
-func (e *blockedEndpoints) unblock(endpoint string) {
+func (e *blockedEndpoints) recover(endpoint string) {
 	e.m.Lock()
 	defer e.m.Unlock()
 
-	delete(e.errorPerEndpoint, endpoint)
+	var b *block
+	if knownBlock, ok := e.errorPerEndpoint[endpoint]; ok {
+		b = knownBlock
+	} else {
+		b = &block{}
```

Comment:
why not exit, recovering from an endpoint never block should do nothing, no ?

Reply:
I think it's useful to keep track of endpoints we hit (also great for testing/debugging).